test/type_construction_test.rb

```diff
@@ -6152,6 +6152,46 @@ def test_logic_receiver_is_nil
     end
   end
 
+  def test_logic_receiver_is_nil_via_annotation
+    with_checker(<<-RBS) do |checker|
+      class Object
+        %a{primitive:nil?}
```

Comment:
Unlike `nil?`, `blank?` doesn't guarantee the receiver is `nil` (e.g. it returns `true` for empty string)